runtime/src/lib.rs

```diff
@@ -28,14 +28,43 @@ pub enum ClientError {
     WentAway = 0xFFFF_FE03,
 }
 
+/// Simple return type that is used when a function can only fail due to
+/// the server dying (i.e. it will never return an error code of its own)
+pub struct ServerDied;
+impl From<ServerDied> for u16 {
```

Comment:
Is this impl necessary? It seems like a bad sign if it is, since this would imply it's being returned to clients from a server.

Reply:
Ah yeah, that was obsoleted by later commits